src/phylo/phyplot.jl

```diff
@@ -0,0 +1,285 @@
+
+    function anglefrompoint(point::Tuple{Number, Number})
+        point[2] >= 0 ? acos(point[1]) : pi + acos(-point[1])
+    end
+
+    function beziercurve(start::Tuple{Number, Number}, angle::Number, center::Tuple{Number, Number})
+        pt = start[1] - center[1], start[2] - center[2]
+        ret = beziercurve(pt, angle)
+        [(pt[1] + center[1], pt[2] + center[2]) for pt in ret]
+    end
+
+    function beziercurve(start::Tuple{Number, Number}, angle::Number)
+        r = sqrt(start[1]^2 + start[2]^2)
+        start = start[1]/r, start[2]/r
+        arc = beziercurve(angle)
+        ret = [rotatepoint(pt, anglefrompoint(start)) for pt in arc]
+        [(r*pt[1], r*pt[2]) for pt in ret]
+    end
+
+    function beziercurve(angle::Number)
+        L  = 4//3*tan(angle/4.)
+        p1 = 1., 0.
+        p2 = 1., L
+        p3 = cos(angle) + L*sin(angle), sin(angle) - L*cos(angle)
+        p4 = cos(angle), sin(angle)
+        [p1, p2, p3, p4]
+    end
+
+    function rotatepoint(point::Tuple{Number, Number}, angle::Number)
+        point[1]*cos(angle) - point[2]*sin(angle), point[1]*sin(angle) + point[2]*cos(angle)
+    end
+
+    function pointcoords(angle::Float64, radius::Float64 = 1.)
+        radius*sin(angle), radius*cos(angle)
+    end
+
+    function normpoint(pt::Tuple{Float64, Float64})
+        pointcoords(pt[2] * circval, pt[1] * radval)
+    end
+
+    function normpoint(ar::Array{Tuple{Float64, Float64},1})
+        [normpoint(x) for x in ar]
+    end
+
+    circval = 2pi/(1 + maximum(values(y)))
+    radval = 1 / maximum(values(x))
+
+    depth = x
+    height = y
+
+    elements = Any[]
+
+    for lin in vertical_lines1
+        bez = beziercurve(normpoint(lin[1]), (lin[1][2] - lin[2][2]) * circval)
+        push!(elements, compose(context(),curve(bez...), stroke(vert1_color)))
+    end
+
+    for lin in vertical_lines2
+        bez = beziercurve(normpoint(lin[1]), (lin[1][2] - lin[2][2]) * circval)
+        push!(elements, compose(context(),curve(bez...), stroke(vert2_color)))
+    end
+
+    rad = [normpoint(x) for x in horizontal_lines]
+    push!(elements, compose(context(),line(rad), stroke(horiz_color)))
+
+    haligns = HAlignment[]
+    valigns = VAlignment[]
+    txts = UTF8String[]
+    rots = Rotation[]
+    pointx = Float64[]
+    pointy = Float64[]
+
+    for tip in terminals
+        point = normpoint((1.03*depth[tip], height[tip]))
+        push!(pointx, point[1])
+        push!(pointy, point[2])
+        push!(valigns, VCenter())
+        push!(haligns, point[1] < 0 ? HRight() : HLeft())
+        push!(txts, tip.name)
+        rot = point[1] < 0 ?  pi + pi/2. - height[tip]*circval : pi/2. - height[tip]*circval
+        push!(rots, Rotation(rot, point))
+    end
+
+    compose(context(0,0,h > w ? w : h, h > w ? w : h, units = UnitBox(-1.6, -1.6, 3.2, 3.2)),
```

Comment:
Ahaa, thanks! I was pulling my hair over the lack of conforming to portrait formats - now I see that the function would always choose ```h```. I will change it to min(w, h), which is also syntactically much clearer.

I am not quite sure about the absolute sizes, but it does sound like it would solve the issues especially with making sure that tip names (when used) are always large enough for people to actually use them. But would a drawback not be that a user might experiment to get the layout perfect in, say, Jupyter, then render to a pdf and realize that all the layout changed? Cannot really tell whether this is a problem or the ideal behaviour.